src/datomish_user_agent_service/api.cljc

```diff
@@ -278,41 +278,32 @@
       (map (fn [[page url title excerpt]]
              {:url url :title title :excerpt excerpt :snippet "" :lastVisited nil})))))
 
-;; TODO: use since.
-;; TODO: return lastVisited, sort by lastVisited.
-;; TODO: support snippet extraction.
 (defn <pages-matching-string [db string {:keys [limit since]
                                          :or {:limit 10}}]
-  (let [string (str "*" string "*")] ;; Wildcard match.  TODO: escape string properly.
+  (let [string (str "*" string "*") ;; Wildcard match.  TODO: escape string properly.
+
+        ;; TODO: extract matching snippet.
+        ;; TODO: return lastVisited, order by lastVisited.
+        ;; TODO: use since, if present.
+        query
+        '[:find [?url (max ?title)]   ; Only take one title, so we're unique on URL.
```

Comment:
Is this max on the value of the title, so some SQLite string ordering?  Should we be maxing over timestamps to get the most recent title?